src/subresource-signing.js

```diff
@@ -8,30 +8,45 @@ var srs = {
      * Load a signed resource at the given url
      * signed in such a way that it can be validated
      * with the given public key.
+     *
+     * IMPORTANT: This only accepts JavaScript files in *.js.asc cleartext signed message format. It does not load the
+     * actual JavaScript file, only clear contents within the *.js.asc file itself if it validates.
+     *
+     * @param armored_pubkey
+     * @param url
+     * @returns {Promise}
      */
-    load: function srs_load(armored_pubkey, url) {
-        var request = new XMLHttpRequest();
-        request.open('GET', url);
-        request.onreadystatechange = function() {
-            if (this.readyState !== 4) return;
-            if (this.status !== 200) {
-                throw srs._.http_error(this.status);
-            }
-            var response = this.response;
-            var promise = srs._.validate(this.response, armored_pubkey);
-            promise.then(function(resolve, reject) {
-                var valid = resolve.signatures.filter(sig => sig.valid).length > 0;
-                if (reject || !valid) {
-                    var message = 'The file at "' + url + '" does not have a valid signature.';
-                    message += ' Refusing to load that file!';
-                    throw message;
+    load: function(armored_pubkey, url) {
+        return new Promise(function(resolve, reject) {
+            var request = new XMLHttpRequest();
+            request.open('GET', url);
+            request.onreadystatechange = function() {
+                if (this.readyState !== 4) return;
+                if (this.status !== 200) {
+                    reject(srs._.http_error(this.status));
+                } else {
+                    var promise = srs._.validate(this.response, armored_pubkey);
+                    promise.then(function(data) {
+                        // Consider the entire response invalid if any of the signatures are not valid.
+                        var valid = data.signatures.filter(sig => sig.valid).length == data.signatures.length;
+                        if (!valid) {
```

Comment:
Say we have a list of trusted pubkeys. Pubkeys are identities. So for example, I could say I trust the jQuery team's pubkey and my own pubkey, but the jQuery file would be signed by that team and my js would be signed by me. Using this validation, neither file would be allowed.